test/tokenbridge.test.ts

```diff
@@ -0,0 +1,204 @@
+  let l2user: Account;
+  let signer: SignerWithAddress;
+  const networkUrl: string = (network.config as HttpNetworkConfig).url;
+  console.log(networkUrl)
+  // L2
+  let L2TokenFactory: StarknetContractFactory;
+  let l2tokenA: StarknetContract;
+  let l2tokenB: StarknetContract;
+  let TokenBridgeL2: StarknetContractFactory;
+  let tokenBridgeL2: StarknetContract;
+  // L1
+  let MockStarknetMessaging: ContractFactory;
+  let mockStarknetMessaging: Contract;
+  let L1TokenFactory: ContractFactory;
+  let l1tokenA: Contract;
+  let l1tokenB: Contract;
+  let TokenBridgeL1: ContractFactory;
+  let tokenBridgeL1: Contract;
+  let messagingContractAddress: string;
+
+  before(async function () {
+
+    // L2 deployments
+
+    l2user = await starknet.deployAccount("OpenZeppelin");
+
+    L2TokenFactory = await starknet.getContractFactory('l2_token');
+    l2tokenA = await L2TokenFactory.deploy(
+        { name: 1234, symbol: 123, decimals: 18, minter_address: BigInt(l2user.starknetContract.address) });
+    l2tokenB = await L2TokenFactory.deploy(
+        { name: 4321, symbol: 321, decimals: 18, minter_address: BigInt(l2user.starknetContract.address) });
+  
+    TokenBridgeL2 = await starknet.getContractFactory('token_bridge');
+    tokenBridgeL2 = await TokenBridgeL2.deploy({ governor_address: BigInt(l2user.starknetContract.address) });
+
+    // L1 deployments
+
+    [signer, l1user] = await ethers.getSigners();
+
+    MockStarknetMessaging = await ethers.getContractFactory(
+      'MockStarknetMessaging',
+      signer,
+    );
+    mockStarknetMessaging = await MockStarknetMessaging.deploy();
+    await mockStarknetMessaging.deployed();
+
+    L1TokenFactory = await ethers.getContractFactory('L1Token', signer);
+    l1tokenA = await L1TokenFactory.deploy(1000);
+    l1tokenB = await L1TokenFactory.deploy(1000);
+
+    TokenBridgeL1 = await ethers.getContractFactory('TokenBridge', signer);
+    tokenBridgeL1 = await TokenBridgeL1.deploy();
+    await tokenBridgeL1.deployed();
+  });
+
+  it('should deploy the messaging contract', async () => {
+    const {
+      address: deployedTo,
+      l1_provider: L1Provider,
+    } = await starknet.devnet.loadL1MessagingContract(networkUrl);
+    expect(deployedTo).not.to.be.undefined;
+    expect(L1Provider).to.equal(networkUrl);    
+  });
+
+  it('should load the already deployed contract if the address is provided', async () => {
+    const {
+      address: deployedTo,
+    } = await starknet.devnet.loadL1MessagingContract(networkUrl);
+
+    await tokenBridgeL1.initializeWithoutProxy(tokenBridgeL2.address, deployedTo);
+    messagingContractAddress = await tokenBridgeL1.messagingContract();
+
+    const {
+      address: loadedFrom,
+    } = await starknet.devnet.loadL1MessagingContract(
+      networkUrl,
+      messagingContractAddress,
+    );
+    expect(messagingContractAddress).to.equal(loadedFrom);
+  });
+
+  it('should exchange messages between L1 and L2', async () => {
```

Comment:
Can we break these tests up into smaller semantic parts and add descriptions in the hardhat sense. Basically break it into smaller `it`s.